cluster/docker/kube-deploy/common.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A common lib for master.sh & node.sh to use
+source ~/docker/kube-config/node.env
+source ~/docker/kube-config/kubelet.env
+
+DESTROY_SH=~/docker/kube-deploy/destroy.sh
+
+: ${K8S_VERSION?"K8S_VERSION is not exported on Node"}
+
+# Run as root
+if [ "$(id -u)" != "0" ]; then
+    echo >&2 "Please run as root"
+    exit 1
+fi
+
+# Make sure docker daemon is running
+if ( ! ps -ef | grep "/usr/bin/docker" | grep -v 'grep' &> /dev/null ); then
+    echo "Docker daemon is not running on this machine!"
+    exit 1
+fi
+
+# Check if a command is valid
+function command_exists() {
+    command -v "$@" > /dev/null 2>&1
+}
+
+# Detect the OS distro, we support ubuntu, debian, mint, centos, fedora dist
+function detect_lsb() {
+    case "$(uname -m)" in
+    *64)
+        ;;
+    *)
+        echo "[ERROR]: We currently only support 64-bit platforms."       
```

Comment:
Just as a note, I'm working on Kubernetes on ARM, so in the future that may be extended to ARM also, as just the docker images have to be switched (maybe some other things too, I don't know yet)
https://github.com/luxas/kubernetes-on-arm